open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/validator/VRPConsentRequestValidator.java

```diff
@@ -0,0 +1,608 @@
+                                PAYLOAD_FORMAT_ERROR_MAXIMUM_INDIVIDUAL_CURRENCY,
+                        ErrorConstants.MAXIMUM_INDIVIDUAL_AMOUNT_CURRENCY_IS_MISSING,
+                        ErrorConstants.PATH_MAXIMUM_INDIVIDUAL_AMOUNT);
+            }
+
+        } else {
+            log.error(ErrorConstants.PAYLOAD_FORMAT_ERROR_MAXIMUM_INDIVIDUAL_AMOUNT);
+            return ConsentManageUtil.getValidationResponse(ErrorConstants.RESOURCE_INVALID_FORMAT,
+                    ErrorConstants.PAYLOAD_FORMAT_ERROR_MAXIMUM_INDIVIDUAL_AMOUNT, ErrorConstants.PATH_REQUEST_BODY);
+
+        }
+        validationResponse.put(ConsentExtensionConstants.IS_VALID, true);
+        return validationResponse;
+    }
+
+    /**
+     * Method to validate variable recurring payment periodic limits.
+     * This method validates the periodic limits specified in the control parameters for variable recurring payments.
+     * It checks if the provided JSON array of periodic limits is valid and then iterates through each limit
+     * to ensure that required fields such as amount, currency, period alignment, and period type are present and
+     * meet the specified criteria.
+     *
+     * @param controlParameters Initiation Object containing periodic limits
+     * @return validation response object indicating whether the provided periodic limits are valid
+     */
+    public static JSONObject validatePeriodicLimits(JSONObject controlParameters) {
+        JSONObject validationResponse = new JSONObject();
+
+        // Check if the periodic limits key is present
+        if (controlParameters.containsKey(ConsentExtensionConstants.PERIODIC_LIMITS)) {
+
+            // Retrieve the periodic limits from the control parameters
+            Object periodicLimit = controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+            // Check if the control parameter is a valid JSON array
+            if (!isValidJSONArray(periodicLimit)) {
+                return ConsentManageUtil.getValidationResponse(ErrorConstants.PAYLOAD_FORMAT_ERROR_PERIODIC_LIMITS,
+                        ErrorConstants.INVALID_PARAMETER_PERIODIC_LIMITS,
+                        ErrorConstants.PATH_PERIOD_LIMIT);
+            }
+
+            // Retrieve the JSON array of periodic limits
+            JSONArray periodicLimits = (JSONArray) controlParameters.get(ConsentExtensionConstants.PERIODIC_LIMITS);
+
+            // Iterate through each periodic limit
+            Iterator<Object> periodicLimitIterator = periodicLimits.iterator();
+
+            while (periodicLimitIterator.hasNext()) {
+                JSONObject limit = (JSONObject) periodicLimitIterator.next();
+
+                // Retrieve values for validation
+
+                Object periodicAlignment = limit.getAsString(ConsentExtensionConstants.PERIOD_ALIGNMENT);
+                Object periodType = limit.getAsString(ConsentExtensionConstants.PERIOD_TYPE);
```

Comment:
Since we are using `getAsString` method rather than `get` method , we can assign the value to type `String` rather than `Object`